server/gokbg3/grails-app/controllers/org/gokb/rest/TitleController.groovy

```diff
@@ -27,8 +27,16 @@ class TitleController {
   def messageService
   def restMappingService
   def titleLookupService
+  def titleHistoryService
   def componentLookupService
 
+  @Secured(['ROLE_USER', 'IS_AUTHENTICATED_FULLY'])
+  def getTypes() {
+    def result = ["serial","monograph","database"]
+
```

Comment:
I'd rather prefer to extract the values from the database, as this is where they reside. If changes appear (new types of media), there will be no need to adapt the implementation and the change becomes effective immediately.

Reply:
You mean checking for existing subclasses of TitleInstance and returning their class name as well as their `niceName`?

Reply:
no, you'e right... I thought, it's a refdata value, like coverage-depth. Since new types of media will implement new subclasses anyway, this is a sufficient solution.